File: Python/001-twoSum.py
 #1 暴力搜索
"""
class Solution:
    def twoSum(self, nums, target):
        """
        :type nums: List[int]
        :type target: int
        :rtype: List[int]
        """
        for m,x in enumerate(nums):
            for n,y in enumerate(nums[m+1:]):
                res = (target-y)
                if x == res:
                    return m,m+1+n
                """

#2 Hash
"""
class Solution:
    def twoSum(self, nums, target):
        buff_dict = {}
        for i in range(len(nums)):
            if nums[i] in buff_dict:
                return [buff_dict[nums[i]], i]
            else:
                buff_dict[target - nums[i]] = i

                """

class Solution:
    def twoSum(self, nums, target):
        dic = {}
        for n,x in enumerate(nums):
            try:
                return dic[x] , n
            except KeyError:
                dic.setdefault(target - x,n)

File: Golang/001-twoSum.go
package main

//1 暴力搜索
/*
func twoSum(nums []int, target int) []int {
	len := len(nums)
	for i := 0; i < len; i++ {
		for j := i + 1; j < len; j++ {
			if nums[i]+nums[j] == target {
				return []int{i, j}
			}
		}
	}
	return []int{}
}
*/
//2 Hash
//  9ms  61.93%
func twoSum(nums []int, target int) []int {
	len := len(nums)
	indexs := make([]int, 2)
	hash := map[int]int{}
	for i := 0; i < len; i++ {
		index, ok := hash[target-nums[i]]
		if ok {
			if i == index {
				continue
			}
			indexs[0] = index
			indexs[1] = i
			break
		} else {
			hash[nums[i]] = i
		}

	}
	return indexs
}
